### PR TITLE
avro-python3 1.10.2 ❄️

### PR DESCRIPTION
**Pull request by psteyer**
avro-python3 1.10.2 ❄️

Destination channel: Snowflake

Links
[JIRA](https://anaconda.atlassian.net/browse/PKG-3811)
[Upstream repository](https://avro.apache.org/)

inspector: https://inspector.pypi.io/project/avro-python3/1.10.2/

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,18 +12,17 @@ source:
...
-  script: {{ PYTHON }} -m pip install . -vv
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -v
 
 requirements:
   host:
-    - python >=3.6
+    - python
     - avro
     - isort
     - pip
```
Comment (lorepirri) [inline recipe/meta.yaml:25]: the usual `setuptools` and `wheel` is required
```suggestion
    - pip
    - setuptools
    - wheel
```

you can see that `setuptools` is imported in the `setup.py` here https://inspector.pypi.io/project/avro-python3/1.10.2/packages/cc/97/7a6970380ca8db9139a3cc0b0e3e0dd3e4bc584fb3644e1d06e71e1a55f0/avro-python3-1.10.2.tar.gz/avro-python3-1.10.2/setup.py#line.39

**Review by lorepirri — COMMENTED** [review]
Please address linter noise:

```
clone/recipe/meta.yaml:39: missing_description: The recipe is missing a description

clone/recipe/meta.yaml:39: missing_license_family: The recipe is missing the about/license_family key.

clone/recipe/meta.yaml:39: missing_documentation: The recipe is missing a doc_url or doc_source_url

clone/recipe/meta.yaml:39: missing_dev_url: The recipe is missing a dev_url

clone/recipe/meta.yaml:23: missing_wheel: For pypi packages, wheel should be present in the host section
```

File: abs.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+upload_channel:
+- sfe1ed40
+channels:
+  - https://staging.continuum.io/prefect/fs/avro-feedstock/pr2/c16245c
```
Comment (boldorider4) [inline abs.yaml]: `avro` has been merged and this line can be removed.

File: recipe/meta.yaml (hunk 1 above)
Comment (boldorider4) [reply recipe/meta.yaml:25]: `setuptools` and `wheel` in general are very common host dependencies for pure python packages.
It is easier to assume setuptools and wheel are always there, except some cases.

**Review by boldorider4 — APPROVED** [review]
LGTM as soon as comments about linter are addressed.

**Comment by skupr-anaconda** [comment]
An error on `win-64`:
```
conda_libmamba_solver.conda_build_exceptions.ExplainedDependencyNeedsBuildingError: Unsatisfiable dependencies for platform win-64: {MatchSpec("avro")}

    Encountered problems while solving:

      - nothing provides requested avro
```

@boldorider4 It's because two `avro`'s tests failed on `win-64` :

```
=========================== short test summary info ===========================
FAILED avro/test/test_script.py::TestCat::test_json_pretty - AssertionError: ...
FAILED avro/test/test_script.py::TestWrite::test_write_csv - subprocess.Calle... 
```
see https://github.com/AnacondaRecipes/avro-feedstock/pull/2

**Comment by boldorider4** [comment]
> @boldorider4 It's because two `avro`'s tests failed on `win-64` :

I'm genuinely wondering why failed builds on win-64 show as skipped and not as failed.

**Comment by skupr-anaconda** [comment]
> > @boldorider4 It's because two `avro`'s tests failed on `win-64` :
> 
> I'm genuinely wondering why failed builds on win-64 show as skipped and not as failed.

It's possible a bug in `conda-build 24.1.[0,1] `

**Comment by lorepirri** [comment]
@psteyer point of action here is: skip `win` please, because `avro` is not available for `win`

File: abs.yaml (hunk 2 above)
Comment (lorepirri) [inline abs.yaml]:
```suggestion
```

**Comment by lorepirri** [comment]
@psteyer only skipping `win` and the linter errors remain to be fixed (note that `win` is `Skipped` but this is an infrastructure (either in `conda-build` or Prefect) bug and it should be `Failed` because `avro` is not available for `win`)

File: recipe/meta.yaml (hunk 1 above)
Comment (lorepirri) [reply recipe/meta.yaml:25]: @psteyer and this one :)

**Review by lorepirri — APPROVED** [review]
LGTM! Thanks for adding all the info!

**Pull request merged**